### PR TITLE
Changed `stdout` and `stderr` from optional to required. Dockerode

### PR DESCRIPTION
**Pull request by tyt4real**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
Docs:
1. https://github.com/apocas/dockerode
2. https://docs.docker.com/engine/api/v1.37/#tag/Container/operation/ContainerLogs

Example:
 - Before:
Code_-_Insiders_5F0342ttCx

 - After:
Code_-_Insiders_QOdCZwiaMm
Code_-_Insiders_CuVr29bYUo

**Basically, if `stdout` or `stderr` are not provided in the options, dockerode will throw an error because one of them must be provided.**

- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/dockerode/dockerode-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 28269ef67407d4f7a71f1958511d5a6f201c745f
</p>

**Comment by nwalters512** [comment]
Could we make the types more precise? Something like this:

```ts
type ContainerLogsOptionsBase = {
    follow?: boolean;
    since?: number | string;
    until?: number | string;
    details?: boolean;
    tail?: number;
    timestamps?: boolean;
    abortSignal?: AbortSignal;
};

type ContainerLogsOptions = 
    | (ContainerLogsOptionsBase & { stdout: true; stderr?: boolean | undefined })
    | (ContainerLogsOptionsBase & { stderr: true; stdout?: boolean | undefined });
```

**Comment by Antonio0806** [comment]
So, I modified the types and tested it both with the command and in a project, I will run the lints in a second

**Comment by jakebailey** [comment]
Why make these required if they have defaults?

**Comment by nwalters512** [comment]
They don't actually have defaults; here's a minimum example of that (run `docker pull ubuntu` first):

```js
const Docker = require('dockerode');
const docker = new Docker();

async function startUbuntuContainer() {
  const container = await docker.createContainer({
    Image: 'ubuntu',
    Cmd: ['/bin/bash', '-c', 'echo Hello from Ubuntu container'],
    Tty: false,
  });
  await container.start();
  await container.wait();

  const logs = await container.logs({
    follow: false,
  });

  console.log(logs.toString('utf8'));
}

startUbuntuContainer().catch((err) => console.error(err));
```

This fails with the following:

```
Error: (HTTP code 400) unexpected - Bad parameters: you must choose at least one stream
```

Adding `stdout: true` to `container.logs(...)` prints `Hello from Ubuntu container` as expected.

Given that at least one must be true, maybe we should instead define the types like this:

```ts
    type ContainerLogsOptions =
        | (ContainerLogsOptionsBase & { stdout: true; stderr?: boolean | undefined })
        | (ContainerLogsOptionsBase & { stderr: true; stdout?: boolean | undefined });
```

**Comment by jakebailey** [comment]
That change certainly seems fine, though potentially a bit annoying to anyone who's dynamically setting these as you are going to need a literal `true` to get it to match. Not super clear to me if this safety is worth it, honestly. If it were me, I'd leave it as-is and document that at least one must be set.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
These seems to be not desirable to make them mandatory despite being optional
